core/src/main/scala/chisel3/ModuleImpl.scala

```diff
@@ -933,8 +936,28 @@ package experimental {
           case Some(c) => getRef.fullName(c)
         }
 
-    /** Returns the current nested module prefix */
-    val modulePrefix: String = Builder.getModulePrefix
+    /** Additional module prefix, applies to this module if defined (unless localModulePrefix is false) and all children.
+      */
+    def localModulePrefix: Option[String] = None
+
+    /** Should [[localModulePrefix]] apply to this module? Defaults to true.
+      *
+      * Users should override to false if [[localModulePrefix]] should apply only to children.
+      */
+    def localPrefixAppliesToSelf: Boolean = true
+
+    /** The resolved module prefix used for this Module.
+      *
+      * Includes [[localModulePrefix]] if defined and if [[localPrefixAppliesToSelf]] is true.
+      */
+    val modulePrefix: String = withModulePrefix(localModulePrefix.filter(_ => localPrefixAppliesToSelf).getOrElse("")) {
```

Comment:
```suggestion
    final val modulePrefix: String = withModulePrefix(localModulePrefix.filter(_ => localPrefixAppliesToSelf).getOrElse("")) {
```
We shouldn't let people mess with this.